docs/release_guide.md

```diff
@@ -121,3 +126,11 @@ Make an announcement on the [mailing list](https://groups.google.com/g/openservi
 Skip this step if the release is a release candidate (RC).
 
 Open a pull request against the `main` branch making the same version updates as [above](#update-release-branch-with-patches-and-versioning-changes) so the latest release assets are referenced there.
+
+## Make version changes on docs.openservicemesh.io
+
+**Note**: do not perform this step for pre-releases.
+
+Associated version numbers need to be updated in the [docs.openservicemesh.io](https://github.com/openservicemesh/osm-docs/) repo.
+
+The suggested way is to search for hard coded current version number in the osm-docs repo, then update the ones that are will brake users' experience with if they follow the documentation, such as demonstration commands, reference links and anything that is strongly related to the next release. We don't need to update the version numbers that just serve the purpose of examples.
```

Comment:
- Some references to code on release branches should be changed, so add a mention to also search for the release branch name
- grammar nits
- add a link to https://github.com/openservicemesh/osm-docs/pull/109 as an example.

```suggestion
The suggested way is to search for hard coded instances of the current version number and release branch name in the osm-docs repo, then update the ones that may break users' experiences if they follow the documentation, such as demonstration commands, reference links and anything that is strongly related to the next release. We don't need to update the version numbers that just serve the purpose of examples.

See https://github.com/openservicemesh/osm-docs/pull/109 for an example of the update from v0.8.4 to v0.9.0.
```